### PR TITLE
Add granular index and constraint recovery info

### LINKED ISSUE #1404: Display which label + property index and which label index has been recreated during snapshot loading
**Memgraph version**
v2.11

**Environment**
Ubuntu

**Describe the bug**
More of a small upgrade to logs -> current message is like the following:
- A label+property index is recreated from metadata.
- A existence constraint is recreated from metadata.
- A unique constraint is recreated from metadata.
- A label index statistics is recreated from metadata.

**To Reproduce**
1. Write a few data points
2. Write some indices and constraints
3. Create snapshot
4. Restart Memgraph
5. Observe the logs

**Expected behavior**
A more granular information for every index and constraint is needed.

### PR DESCRIPTION
**Pull request by Josipmrden**
Index and constraint recovery info was not very descriptive.

closes #1404 

[master < Task] PR
- [x] Check, and update documentation if necessary
- [x] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [x] Write a release note here, including added/changed clauses
- [x] Tag someone from docs team in the comments

@vpavicic : Changelog message: Memgraph is now showing which indices and constraints is recovering in the logs.

### CODE REVIEW

File: src/storage/v2/durability/durability.cpp [hunk 1]
```diff
@@ -203,7 +209,14 @@ void RecoverIndicesAndConstraints(const RecoveredIndicesAndConstraints &indices_
     auto ret = mem_unique_constraints->CreateConstraint(item.first, item.second, vertices->access());
     if (ret.HasError() || ret.GetValue() != UniqueConstraints::CreationStatus::SUCCESS)
       throw RecoveryFailure("The unique constraint must be created here!");
-    spdlog::info("A unique constraint is recreated from metadata.");
+
+    std::vector<std::string> property_names;
```
Comment (as51340) [inline src/storage/v2/durability/durability.cpp:213]:
```suggestion
    std::vector<std::string> property_names;
    property_names.reserve(item.second.size())
```

File: src/storage/v2/durability/durability.cpp [hunk 2]
```diff
@@ -203,7 +209,14 @@ void RecoverIndicesAndConstraints(const RecoveredIndicesAndConstraints &indices_
...
+    for (const auto &prop : item.second) {
+      property_names.push_back(name_id_mapper->IdToName(prop.AsUint()));
```
Comment (as51340) [inline src/storage/v2/durability/durability.cpp:215]:
```suggestion
      property_names.emplace_back(name_id_mapper->IdToName(prop.AsUint()));
```

File: src/storage/v2/durability/durability.cpp [hunk 3]
```diff
@@ -31,10 +31,12 @@
 #include "storage/v2/inmemory/label_index.hpp"
 #include "storage/v2/inmemory/label_property_index.hpp"
 #include "storage/v2/inmemory/unique_constraints.hpp"
+#include "storage/v2/name_id_mapper.hpp"
 #include "utils/event_histogram.hpp"
 #include "utils/logging.hpp"
 #include "utils/memory_tracker.hpp"
 #include "utils/message.hpp"
+#include "utils/string.hpp"
```
Comment (as51340) [inline src/storage/v2/durability/durability.cpp:39]: You probably don't need either of these includes?

**Review by as51340 — CHANGES_REQUESTED** [review]

File: src/storage/v2/durability/durability.cpp (hunk 3 above)
Comment (Josipmrden) [reply src/storage/v2/durability/durability.cpp:39]: probably not, I can't recall why it was added. Maybe just automatically.

File: src/storage/v2/durability/durability.cpp [hunk 4]
```diff
@@ -203,7 +209,14 @@ void RecoverIndicesAndConstraints(const RecoveredIndicesAndConstraints &indices_
     auto ret = mem_unique_constraints->CreateConstraint(item.first, item.second, vertices->access());
     if (ret.HasError() || ret.GetValue() != UniqueConstraints::CreationStatus::SUCCESS)
       throw RecoveryFailure("The unique constraint must be created here!");
-    spdlog::info("A unique constraint is recreated from metadata.");
+
+    std::vector<std::string> property_names;
+    for (const auto &prop : item.second) {
+      property_names.push_back(name_id_mapper->IdToName(prop.AsUint()));
+    }
+    auto property_names_joined = utils::Join(property_names, ",");
```
Comment (antoniofilipovic) [inline src/storage/v2/durability/durability.cpp:217]:
```suggestion
    const auto property_names_joined = utils::Join(property_names, ",");
```

File: src/storage/v2/durability/durability.hpp [hunk 5]
```diff
@@ -104,7 +104,7 @@ using ParallelizedIndexCreationInfo =
 /// @throw RecoveryFailure
 void RecoverIndicesAndConstraints(
     const RecoveredIndicesAndConstraints &indices_constraints, Indices *indices, Constraints *constraints,
-    utils::SkipList<Vertex> *vertices,
+    utils::SkipList<Vertex> *vertices, NameIdMapper *name_id_mapper,
```
Comment (antoniofilipovic) [inline src/storage/v2/durability/durability.hpp:107]:
```suggestion
    utils::SkipList<Vertex> *vertices, const NameIdMapper *name_id_mapper,
```

Comment (antoniofilipovic) [reply src/storage/v2/durability/durability.hpp:107]: Since you are not inserting anything here

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Few comments

**Review by gitbuda — APPROVED** [review]

**Pull request merged**